File: 0048.旋转图像/0048-旋转图像.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int l = matrix[0].size();
        for (int i = 0 ; i < matrix.size(); ++i){
            for (int j = i + 1; j < l; ++j)
                swap(matrix[i][j], matrix[j][i]);
            reverse(matrix[i].begin(), matrix[i].end());
        }
        
    }
};

File: LeetCode-Python/0048.旋转图像/0048-旋转图像.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        n = len(matrix)
        for i in range(n):
            for j in range(i + 1, n):
                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]

        for i in range(n):
            row = matrix[i]
            for j in range(n // 2):
                row[j], row[-(j + 1)] = row[-(j + 1)], row[j]
        
